### ISSUE TITLE
Feature proposal: Option to turn off automatic chunk naming

### ISSUE THREAD

**Issue by Ephem** [gregberge/loadable-components#740]
## 🚀 Feature Proposal

Add a new option to the Babel plugin to turn off automatic chunk naming.

## Motivation

The Babel plugin currently does two things, add SSR support and automatically add chunk naming via a magic Webpack comments. In some cases, you might only want to add the SSR support, but not add readable names to the chunks.

For example, with Webpack's split chunks, the chunk names will also affect the file names, and you might not want to expose the names of the chunks to the end users for various reasons (I also _think_ having chunk names can in some cases affect _how_ Webpack splits chunks). Even with Webpack 5s prod-default 'deterministic' option will still preserve these chunk names because they were added via a magic comment.

## Example

`new LoadablePlugin({ automaticChunkNames: false })`

## Pitch

There are valid cases where you don't want automatic chunk names and this feature makes sense/can only be implemented inside of Loadable itself. It's also a minimal API to both implement and maintain in the future, so no real downsides. This API also enables having chunk names in dev, but avoiding them in prod. Keeping the default to `true` makes this a non-breaking change.

If there is interest I could probably find time to try implementing this.

**Comment by theKashey** [comment]
Well, it's not about chunk naming, its about the ability to find the right chunk to load when needed.
Especially to keep files extracted in a way expected by loadable components - a chunk per import.

There is no way we can disable automatic chunk naming without deep refactoring of `webpack plugin` first - technically it's possible to _connect_ import and the corresponding chunk by the entry filename, but this logic is simply missing right now.

**Comment by Ephem** [comment]
Thanks for elaborating! This doesn't change what you are saying, but I realize now that I totally botched the API design, I meant to add the configuration option to the _Babel_ plugin, not the Webpack one, since that's the one adding the comments. So more like:

```
{
  "plugins": [["@loadable/babel-plugin", { automaticChunkNames: false }]]
}
```

I've dug some more into the src and I think I understand what you mean now, but just to see if I get it, this is roughly my mental model of what happens:

* The Babel plugin creates a name for the chunk
  * When it remakes the `() => import(...)` pattern to an object instead, it places that name on that object (accessed via `.chunkName()`)
  * It also adds this name as a magic Webpack comment _with the explicit purpose to make Webpack name it that so it matches the info we have available in the runtime code_
* On the server, the component created by `loadable()` calls `ChunkExtractor.addChunk()` to indicate it needs to be loaded
* When creating tags, the `ChunkExtractor` looks up the names of the added chunks in the stats-file

So in other words, the Babel plugin is connecting/shaping the chunkName for both the buildtime (Webpack comment) and runtime (`.chunkName()`) together. Automatic naming is not a feature, it's a way of implementation?

If I understood this correctly (?) it's definitely trickier than I imagined and even though possible I can see how it's not a priority given the scope. 😄

**Comment by theKashey** [comment]
Yes, this is more about implementation detail rather than a feature. 

It's still not a big problem to find a chunk name from the stats. It's just not how it's working right now, and it might be not that easy to change this.

From time to time I am thinking about doing a major change with some necessary API updates, and that can be a part of it. One day.

**Comment by Ephem** [comment]
Sounds great!

This feature was not super important for us as we don't mind exposing the chunk names in prod currently. I'll leave the decision of closing this or keeping it open to track for the future to you, I don't mind either way. 😄 

Thanks again for all your hard work on this library!
